schedule/schedule-appengine/src/main/java/com/jasify/schedule/appengine/dao/common/ActivityPackageActivityDao.java

```diff
@@ -45,6 +84,60 @@ public ActivityPackageActivityDao() {
         return query(new ByOrganizationAndActivityQuery(meta, organizationId, activityId));
     }
 
+    public Key getKeyByActivityPackageIdAndActivityId(Key activityPackageId, Key activityId) throws EntityNotFoundException {
+        ActivityPackageActivityMeta meta = getMeta();
+
+        ActivityPackage activityPackage = activityPackageDao.get(activityPackageId);
+        Key organizationId = activityPackage.getOrganizationRef().getKey();
+
+        List<Key> result = queryKeys(new ByActivityPackageAndActivityQuery(meta, organizationId, activityPackageId, activityId));
+
+        // There can be only one
+        if (result.isEmpty()) {
+            return null;
+        } else if (result.size() > 1) {
+            throw new PreparedQuery.TooManyResultsException();
+        }
```

Comment:
ModelException extends Exception so this must not be possible as it would force declaration. TooManyResultsException is a RunTimeException so this is possible but none of the methods being called seem to contain this exception and we are not generally catching RunTimeExceptions so I am not sure.